tools/topology/topology1/CMakeLists.txt

```diff
@@ -0,0 +1,216 @@
+	"sof-tgl-max98373-rt5682\;sof-tgl-max98373-rt5682-xperi\;-DAMP_SSP=1\;-DINCLUDE_IIR_EQ=1"
+	"sof-tgl-max98373-rt5682\;sof-tgl-rt5682-ssp0-max98373-ssp2\;-DAMP_SSP=2"
+	"sof-tgl-max98373-rt5682\;sof-tgl-rt5682-ssp0-max98373-ssp2-xperi\;-DAMP_SSP=2\;-DINCLUDE_IIR_EQ=1"
+	"sof-tgl-sdw-max98373-rt5682\;sof-tgl-sdw-max98373-rt5682\;-DCHANNELS=4\;-DPLATFORM=tgl"
+	"sof-tgl-sdw-max98373-rt5682\;sof-adl-sdw-max98373-rt5682\;-DCHANNELS=4\;-DPLATFORM=adl"
+	"sof-jsl-da7219\;sof-jsl-da7219\;-DPLATFORM=jsl"
+	"sof-jsl-da7219\;sof-jsl-da7219-mx98360a\;-DPLATFORM=jsl-dedede"
+	"sof-imx8mp-wm8960\;sof-imx8mp-wm8960"
+	"sof-smart-amplifier-nocodec\;sof-smart-amplifier-nocodec"
+	"sof-jsl-rt5682\;sof-jsl-rt5682-rt1015\;-DPLATFORM=jsl-rt1015"
+	"sof-jsl-rt5682\;sof-jsl-rt5682-rt1015-xperi\;-DPLATFORM=jsl-rt1015\;-DINCLUDE_IIR_EQ=1"
+	"sof-jsl-rt5682\;sof-jsl-rt5682-mx98360a\;-DPLATFORM=jsl-dedede"
+)
+
+add_custom_target(topologies1_output ALL)
+add_custom_target(topologies1 ALL
+    COMMAND ${CMAKE_COMMAND} -E copy_directory . ${SOF_TOPOLOGY_BINARY_DIRECTORY}
+    DEPENDS topologies1_output
+)
+
+foreach(tplg ${TPLGS})
+	list(GET tplg 0 input)
+	list(GET tplg 1 output)
+	list(LENGTH tplg tplg_len)
+	math(EXPR num_args "${tplg_len} - 1")
+	set(DEFINES "")
+	if (${num_args} GREATER 1)
+		foreach(index RANGE 2 ${num_args})
+			list(GET tplg ${index} custom_define)
+			list(APPEND DEFINES ${custom_define})
+		endforeach()
+	endif()
+
+	add_custom_command(
+		OUTPUT ${output}.conf
+		COMMAND m4 --fatal-warnings
+			${DEFINES}
+			-I ${CMAKE_CURRENT_SOURCE_DIR}/m4
+			-I ${CMAKE_CURRENT_SOURCE_DIR}/common
+			-I ${CMAKE_CURRENT_SOURCE_DIR}/platform/common
+			-I ${CMAKE_CURRENT_SOURCE_DIR}
+			-I ${CMAKE_CURRENT_BINARY_DIR}
+			${CMAKE_CURRENT_SOURCE_DIR}/common/abi.m4
+			${CMAKE_CURRENT_SOURCE_DIR}/${input}.m4
+			> ${output}.conf
+		MAIN_DEPENDENCY ${CMAKE_CURRENT_SOURCE_DIR}/${input}.m4
+		DEPENDS abi
+		VERBATIM
+		USES_TERMINAL
+	)
+
+# This use of VERBOSE relies on original CMake behavior.
+# From the add_custom_command() manual:
+#
+#    Use of VERBATIM is recommended as it enables correct behavior.
+#    When VERBATIM is not given the behavior is platform specific because
+#    there is no protection of tool-specific special characters.
+#
+# This is fine because:
+# - We don't expect alsatplg to work on Windows any time soon.
+# - CMake is too afraid to remove the original, no-VERBATIM behavior:
+#      https://gitlab.kitware.com/cmake/cmake/issues/18849
+#
+# Also note that in alsa-utils commit v1.2.2~15-gcbabe7a3f0cc, alsatplg
+# (accidentally?) started ignoring the verbosity level, now it's just
+# verbose or not.
+	add_custom_command(
+		OUTPUT ${output}.tplg
+		COMMAND alsatplg \$\${VERBOSE:+-v 1} -c ${CMAKE_CURRENT_BINARY_DIR}/${output}.conf -o ${output}.tplg
+		MAIN_DEPENDENCY ${output}.conf
+		USES_TERMINAL
+	)
+
+	add_custom_target(topologies1_${output} DEPENDS ${output}.tplg)
+	add_dependencies(topologies1_output topologies1_${output})
```

Comment:
The previous code used singular "topology" for a single file and plural "topologies" for the target that depends on them all. I think this made it very clear that the `add_dependencies(topologies topology_${i})` line is growing a list, even for people who don't know anything about CMake. Can you please preserve that?